headers.go

```diff
@@ -471,6 +497,25 @@ func validateHeaderParameters(h map[any]any, protected bool) error {
 			if err := ensureCritical(value, h); err != nil {
 				return fmt.Errorf("header parameter: crit: %w", err)
 			}
+		case HeaderLabelType:
+			is_tstr := canTstr(value)
+			if !is_tstr && !canUint(value) {
+				return errors.New("header parameter: type: require tstr / uint type")
+			}
+			if is_tstr {
+				v := value.(string)
+				if len(v) == 0 {
+					return errors.New("header parameter: type: require non-empty string")
+				}
+				if v[0] == ' ' || v[len(v)-1] == ' ' {
+					return errors.New("header parameter: type: require no leading/trailing whitespace")
+				}
+				// Basic check that the content type is of form type/subtype.
+				// We don't check the precise definition though (RFC 6838 Section 4.2).
+				if strings.Count(v, "/") != 1 {
+					return errors.New("header parameter: type: require text of form type/subtype")
+				}
```

Comment:
this looks worth refactoring (see also L524-537).

Reply:
I agree, but I am not really a go dev... probably better for someone with better intuition to attempt it

Reply:
No worries, we can refactor it later.